### PR TITLE
Add a windows installer script

### PR DESCRIPTION
**Pull request by mhogomchungu**
This pull requests adds an "inno setup"[1] installer script to the project. This script will make it possible to easily package and distribute encfs for windows using an installer.

[1] http://www.jrsoftware.org/isinfo.php

### CODE REVIEW

File: windows-installer.iss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+; SEE THE DOCUMENTATION FOR DETAILS ON CREATING INNO SETUP SCRIPT FILES!
+
+#define MyAppName "Encfs"
+#define MyAppVersion "1.9.5"
+#define MyAppPublisher "encfs"
+#define MyAppURL "https://github.com/vgough/encfs"
+#define MyAppExeName "encfs.exe"
+
+; This variable should contain a path to where libraries needed by encfs are installed.
+#define MyLibraryPath "C:\cygwin\usr\bin"
```
Comment (benrubson) [inline windows-installer.iss]: Perhaps drop the `usr` part of the path ?
As the real Windows path is `C:\cygwin\bin`, `C:\cygwin\usr\bin` being a mount point done by Cygwin to `C:\cygwin\bin`.

File: windows-installer.iss [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+; This variable should contain a path to where cygwin binaries to be packaged are located.
+#define MyEXEPath "C:\cygwin\usr\bin"
```
Comment (benrubson) [inline windows-installer.iss]: Same here, perhaps drop the `usr` part of the path which may be not available out of Cygwin itself ?
Thx 👍

File: windows-installer.iss (hunk 1 above)
Comment (mhogomchungu) [reply windows-installer.iss]: Done on both recommendations.

Comment (benrubson) [reply windows-installer.iss]: Thank you 👍

**Comment by benrubson** [comment]
Is assume this script is rather dedicated to the 32 bits version ?
(as Cygwin 64 bits is by default installed into `C:\cygwin64`)

We could also simply assume to install the 64 bits version into `C:\cygwin`...

Sounds like Appveyor has both 32 and 64 bits Cygwin versions built-in :
https://www.appveyor.com/docs/build-environment/#mingw-msys-cygwin
We could simply rename `C:\cygwin64` to `C:\cygwin` in an automated build scenario.

**Comment by mhogomchungu** [comment]
I think a decision will have to be made here.

1. Ship only 32 bit version of the installer because it works with both 32 bit and 64 bit windows.

2. Ship installers for both 32 bit and 64 bit installers and let users pick the one they prefer.

If we go with the former, nothing needs to be done, if with the latter then we could do the folder substitution trick as suggested or have a separate installer script for 64 bit and name it differently to differentiate the two. Having two installer script will make it easier to build both installers with the same CI instance if it will be possible to have versions of cygwin installed.

**Comment by benrubson** [comment]
Because of the issue we discussed [here](https://github.com/vgough/encfs/issues/508#issuecomment-384597646), I think we need at least a 64 bits version, so that the 32 bits version should be chosen by users running a 32 bits OS only.

> Having two installer script will make it easier to build both installers with the same CI instance if it will be possible to have versions of cygwin installed.

Yes you're right. A little bit (just a little) more difficult to maintain though, if we need to update them (did you update both scripts ?).
According to what I understand [from this](https://www.appveyor.com/docs/build-environment/#mingw-msys-cygwin), both 32 and 64 bits Cygwin versions are installed in all AppVeyor images.

**Comment by billziss-gh** [comment]
FWIW, my approach has been to package all necessary Cygwin files within the installer in [SSHFS-Win](https://github.com/billziss-gh/sshfs-win) and [NFS-Win](https://github.com/billziss-gh/nfs-win).

**Comment by benrubson** [comment]
We should too, and according to https://github.com/billziss-gh/winfsp/issues/161, sounds like it could be important to test and choose Cygwin version to package !

**Comment by mhogomchungu** [comment]
So, whats the verdict on the pull request?

This is as far as i can take take the installer script. Decisions on how many packages to build and from where could be handled internally later on.

**Comment by rfjakob** [comment]
Thanks for the installer script! I have merged it into the new "cygwin" branch as 54f8aa1 . We should experiment there, get the build infrastructure right, and squash and merge into master when ready.

**Pull request closed without merging**